globus_cli/commands/endpoint/server/list.py

```diff
@@ -14,6 +17,28 @@ def server_list(endpoint_id):
     """
     client = get_client()
 
+    endpoint = client.get_endpoint(endpoint_id)
+
+    if endpoint['host_endpoint_id']:  # not GCS -- this is a share endpoint
+        raise click.UsageError(
+            "{id} ({display_name}) is a share and does not have servers.\n"
```

Comment:
I tested this against a Share with no `display_name` set.
Can we use `globus_cli.services.transfer.display_name_or_cname`?

You can do it with minimal noise if you add it under a name that won't conflict or as a positional arg to `format`.
Positional is probably safer.